### DISCUSSION TITLE
Attempting to create a new file with different extension

### DISCUSSION THREAD

**Discussion by sawft99** [SilentVoid13/Templater#1311] [Help]
My code:

```
<%*
let ProjectName = await tp.system.prompt("Project name")
let MaterialsName = ProjectName + " Materials"
let StepsName = ProjectName + " Steps"
await tp.file.rename(ProjectName)
let MaterialsLoomTemplate = tp.file.find_tfile("Template Loom Materials.loom") 
await tp.file.create_new(MaterialsLoomTemplate, MaterialsName + ".loom", false, app.vault.getAbstractFileByPath("Attachments"))
let StepsLoomTemplate = tp.file.find_tfile("Template Loom Steps.loom") 
await tp.file.create_new(StepsLoomTemplate, StepsName + ".loom", false, app.vault.getAbstractFileByPath("Attachments"))
-%>
```

I'm attempting to create a file with from a template, but the template file and the destination file should end with `.loom`.
When the line `await tp.file.create_new(StepsLoomTemplate, StepsName + ".loom", false, app.vault.getAbstractFileByPath("Attachments"))` runs, it just makes a file with `.loom` in the name and doesn't actually output to a file with that extension.

**Comment by AB1908** [comment]
What is the use case?

**Reply by sawft99** [reply]
...making a new loom file based on a template with a loom extension as well. 

If you need more information. The loom file is part of the dataloom plugin. I'm attempting to trigger the creation of 2 accompanying loom files for every project file i make. One for materials and one for steps.

**Reply by AB1908** [reply]
Out of curiosity, have you tried using Obsidian's API to create the file directly? Also, does the dataloom extension provide an API to do this? Maybe you could ask the author to expose one to make the process smoother and more generally usable.

**Reply by sawft99** [reply]
If I were any kind of real developer maybe. I was hoping templater could do this, I have everything in place and working. It just needs to be a .loom file instead a .md when created. The process seems smooth enough.

**Comment by Zachatoo** [comment]
@sawft99 Should now be possible in Templater 2.2.0. Do not include the extension when calling `tp.file.create_new()`, Templater will use the same extension as the template file you're creating from. For example, because MaterialsLoomTemplate is a `.loom` file, if you call `tp.file.create_new(MaterialsLoomTemplate, ...)`, the created file will also be a `.loom` file. Templater will also attempt to execute any Templater commands in that `.loom` file.

**Reply by sawft99** [reply]
Awesome. Thank you.
